### PR TITLE
[WIP] Issue #1262 - Option to disable Auto-Enclosing Tags

### LINKED ISSUE #1262: Provide option to disable auto-enclosing tags
From a tweet: https://twitter.com/agentrockstar11/status/666887017356095488

### PR DESCRIPTION
**Pull request by cgsingh**
In correlation with: https://github.com/mozilla/brackets/pull/601

I finally got it down to 1 commit. I don't know why it wouldn't let me squash my merge commit into another commit.

Fixes #1262

### CODE REVIEW

**Comment by flukeout** [comment]
Ping @cgsingh - are you able to make the changes requested by @gideonthomas? Once you do, I'll test it out and we can get it merged if everything checks out.

Thanks!

**Comment by cgsingh** [comment]
@flukeout I pushed the changes, but now I have to look at the brackets PR and make the changes requested there.

**Comment by cgsingh** [comment]
@humphd @flukeout Unsurprisingly, it looks like there is a new problem. On initial load, closeTags is supposed to be enabled by default. It *is* according to getAutoCloseTags() which returns the object that states it's enabled. However, that doesn't reflect in the editor.

As a temporary fix in public/editor/scripts/editor/js/fc/bramble-menus.js on line 86, I have it call the configure method. But I highly doubt you want it calling that every time the toggle is enabled...

**Comment by flukeout** [comment]
Hey @cgsingh - I just tested this along with the Brackets PR and everything seems to be working well. Are you still working on this and making changes?

**Comment by cgsingh** [comment]
@flukeout Hi. No, I'm not making changes on this particular PR. But I've just finished making the changes requested on the Brackets PR. I think (I hope) it's good to go.

**Comment by flukeout** [comment]
Okay, thanks - testing now @cgsingh

File: locales/en-US/messages.properties [hunk 1]
```diff
@@ -102,6 +102,7 @@ navTextSize=Text size
 navColorTheme=Color theme
 navWrapText=Wrap long lines
 navAllowJS=Allow JavaScript
+navAutoEncloseTags=Auto-enclose tags
```
Comment (flukeout) [inline locales/en-US/messages.properties]: Please change to "Auto close tags"

File: locales/en-US/messages.properties [hunk 2]
```diff
@@ -196,6 +197,7 @@ navDarkThemeIconTitle=Switch to the dark theme
 navLightThemeIconTitle=Switch to the light theme
 navWrapTextTitle=Toggle wrapping long lines of code
 navAllowJSTitle=Toggle running JavaScript code in the preview
+navAutoEncloseTagsTitle=Toggle automatically enclosing tags
```
Comment (flukeout) [inline locales/en-US/messages.properties]: Please change to "Toggle automatically closing HTML tags"

**Comment by cgsingh** [comment]
@flukeout changed it

**Comment by flukeout** [comment]
Great, will test it now, thanks!

**Comment by flukeout** [comment]
Awesome - works great and remembers if it was disabled or enabled after refreshing the editor. Great work @cgsingh 👍

**Comment by flukeout** [comment]
Hey @cgsingh - is there an email I can reach you at? I'm collecting information for a contributors page on the Thimble website and I'd like to include you. Can you please email me at ``luke@mozillafoundation.org``? Thanks!

**Pull request merged**